src/libcore/macros/mod.rs

```diff
@@ -6,9 +6,13 @@ macro_rules! panic {
     () => (
         $crate::panic!("explicit panic")
     );
-    ($msg:expr) => (
+    ($msg:literal) => (
         $crate::panicking::panic($msg)
     );
+    ($msg:expr) => ({
+        let msg: &str = $msg; // Make sure it is a &str.
+        $crate::panic!("{}", msg)
+    });
     ($msg:expr,) => (
         $crate::panic!($msg)
     );
```

Comment:
You should do the same thing for the trailing comma case.

Reply:
The `($msg:expr, )` case already forwards that to the `($msg:literal)` case: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=00ab9b02d3d215e0f920c1c94047dc5c

Reply:
Ah, my bad, misread the code.